### PR TITLE
#7690 Add Playwright Extension Console smoke test

### PR DESCRIPTION
**Pull request by mnholtz**
## What does this PR do?

- Part of #7690
- Adds a fixture for running the browser with a build of the PixieBrix extension
- Makes some adjustments to our auth utility in order get it to work with the fixture
- Adds a smoke test that asserts that the Extension Console is rendered properly

## Reviewer Tips

- Pull this branch down and run `npm run test:e2e` to see it in action 🙂

## Checklist

- [x] Add tests
- [x] New files added to `src/tsconfig.strictNullChecks.json` (if possible)
- [x] Designate a primary reviewer @fungairino

### CODE REVIEW

**Comment by twschiller** [comment]
image

**Comment by twschiller** [comment]
FWIW, I'm getting:

image

File: playwright.config.ts [hunk 1]
```diff
@@ -24,6 +24,8 @@ export default defineConfig({
 
     /* Collect trace when retrying the failed test in CI, and always on failure when running locally. See https://playwright.dev/docs/trace-viewer */
     trace: CI ? "on-first-retry" : "retain-on-failure",
+    // Use auth state prepared by auth.setup.ts
+    storageState: "end-to-end-tests/.auth/user.json",
```
Comment (fungairino) [inline playwright.config.ts:28]: I think with this change playwright expects this file to be present on start-up, so we should revert the change in this file specifically.

We might also want to rename this path to `.storage/user.json` since this is where we will modify other stuff in the session.

File: end-to-end-tests/fixtures.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    const fileBuffer = await fs.readFile(
+      // eslint-disable-next-line unicorn/prefer-module -- TODO: import.meta.dirname is throwing "cannot use 'import meta' outside a module"
+      path.join(__dirname, "./.auth/user.json"),
+    );
+
+    const { cookies } = JSON.parse(fileBuffer.toString()) as unknown as {
+      cookies: Cookie[];
+    };
+
+    // Manually add session cookies instead of relying on storageState in playwright.config.ts because
```
Comment (fungairino) [inline end-to-end-tests/fixtures.ts:53]: We should link the related playwright github issue we found.

File: end-to-end-tests/fixtures.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+} from "@playwright/test";
+import path from "node:path";
+import { MV } from "./env";
+import fs from "node:fs/promises";
+
+export const test = base.extend<{
+  context: BrowserContext;
+  extensionId: string;
+}>({
+  // eslint-disable-next-line no-empty-pattern -- Playwright requires destructuring pattern as first argument
```
Comment (fungairino) [inline end-to-end-tests/fixtures.ts:32]: nit: we should consider ignoring some rules for this e2e directory in the eslint config.

File: end-to-end-tests/example.spec.ts [hunk 4]
```diff
@@ -26,6 +26,11 @@ test.describe("create-react-app", () => {
     ).toBeVisible();
   });
 
+  test("can open the extension console", async ({ page, extensionId }) => {
+    await page.goto(`chrome-extension://${extensionId}/options.html`);
+    await expect(page.getByText("Extension Console")).toBeVisible();
```
Comment (fungairino) [inline end-to-end-tests/example.spec.ts:31]: as a fast follow-up let's check if the user is authenticated when opening the extension console.

**Review by fungairino — COMMENTED** [review]
Looks good. We just need to fix the storage issue

File: playwright.config.ts (hunk 1 above)
Comment (mnholtz) [reply playwright.config.ts:28]: > since this is where we will modify other stuff in the session

Like what?

File: end-to-end-tests/example.spec.ts (hunk 4 above)
Comment (mnholtz) [reply end-to-end-tests/example.spec.ts:31]: Oops, I was wrong! 
Screen Shot 2024-03-08 at 10 46 06 AM

File: playwright.config.ts (hunk 1 above)
Comment (fungairino) [reply playwright.config.ts:28]: cookies for other purposes and local storage

**Review by fungairino — APPROVED** [review]

**Comment by twschiller** [comment]
Wooooo
image

**Comment by mnholtz** [comment]
Woo hoo, we fixed CI
Screen Shot 2024-03-08 at 2 36 59 PM

**Review by fungairino — APPROVED** [review]

**Review by fungairino — APPROVED** [review]
